runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6484,7 +6484,7 @@ void
 TR_J9ByteCodeIlGenerator::genMonitorExit(bool isReturn)
    {
    TR::SymbolReference * monitorExitSymbolRef =
-      isReturn && isOutermostMethod() ?
+      isReturn && (isOutermostMethod() || comp()->getOption(TR_EnableOSR)) ?
```

Comment:
Similarly here, isn't isReturn sufficient to select the symbol?

Reply:
changed accordingly.